lib/types/api.d.ts

```diff
@@ -0,0 +1,41 @@
+interface IOption<T> {
+    [key: string]: T
+}
+
+interface IPayload {
+    url: string,
+    params: IOption<any>
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏Make IPayload a Generic and pass that to a params, lets not use IOption here

Reply:
Can we pass options to each API call?

Reply:
Done